src/deploy/functions/validate.ts

```diff
@@ -126,6 +127,23 @@ function assertFunctionsSourcePresent(data: any, sourceDir: string, projectDir:
   }
 }
 
+/**
+ * Asserts that package.json does not include any scripts with invalid names
+ * @param data Object representing package.json file.
+ * @throws { FirebaseError } package.json must not contain scripts with protected names
+ */
+function checkForProtectedNpmScriptNames(data: any): void {
+  const protectedScriptNames = ["prepare"];
+  if (Object.prototype.hasOwnProperty.call(data, "scripts")) {
+    for (const scriptName of protectedScriptNames) {
+      if (Object.prototype.hasOwnProperty.call(data.scripts, scriptName)) {
```

Comment:
```suggestion
  if (data?.scripts) {
    for (const scriptName of protectedScriptNames) {
      if (data.scripts.[scriptName]) {
```

@joehan Just getting your first suggestion but keeping the "fail fast" approach of that first conditional so it doesn't start the for loop unnecessarily. What do you think?

Reply:
Seems reasonable to me